packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -162,30 +150,56 @@ export const render = async (req, res, next) => {
     const component = await route.component.getComponent()
 
     // Step 3 - Init the app state
-    const {appState, error: appStateError} = await initAppState({
-        App: WrappedApp,
-        component,
-        match,
-        route,
-        req,
-        res,
-        location
-    })
+    let appState = {}
+    let routerContext = {}
+    let error
+    let html
+    let appStateError
+
+    // Step 3. Bail if there is a 404.
+    if (component === Throw404) {
+        error = new errors.HTTPNotFound('Not found')
+    }
+
+    if (!error) {
+        const AppJSX = getAppJSX(req, res, error, {
+            App: WrappedApp,
+            appState,
+            deviceType,
+            location,
+            routerContext,
+            routes
+        })
+
+        ;({appState, error: appStateError} = await WrappedApp.fetchState({
```

Comment:
Dumb question, why do we need to wrap this line around `;( ... )` ?

Reply:
This line isn't in the code anymore, but for education purposes. I found out that this is how you accomplish using let to declare variables and have them be assigned to during destructuring. 

The reason I was having to do this is because `appState` and `appStateError` were declared outside of this conditional because they are being used elsewhere. Alternatively you could use an intermediate variable like this

```
const result = await WrappedApp.fetchState(...)
appState = result.appState
appStateError = result.appStateError
```